hydromt/data_adapter/dataframe.py

```diff
@@ -143,8 +149,18 @@ def get_data(
         For a detailed description see: :py:func:`~hydromt.data_catalog.DataCatalog.get_dataframe`
         """
 
+        # Extract storage_options from kwargs to instantiate fsspec object correctly
+        if "storage_options" in self.kwargs:
+            kwargs = self.kwargs["storage_options"]
+            # For s3, anonymous connection still requires --no-sign-request profile to read the data
+            # setting environment variable works
+            if "anon" in kwargs:
+                os.environ["AWS_NO_SIGN_REQUEST"] = "YES"
+            else:
+                os.environ["AWS_NO_SIGN_REQUEST"] = "NO"
```

Comment:
perhaps this could eventually be generalized in storage_options method of the DataAdapter class